olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Versions.h

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <olp/dataservice/read/DataServiceReadApi.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+namespace model {
+
+/**
+ * @brief Represents a catalog version dependencies
+ */
+class DATASERVICE_READ_API CatalogVersionDependencies final {
+ public:
+  /**
+   * @brief Sets the hrn.
+   *
+   * @param hrn.
+   */
+  void SetHrn(std::string hrn) { hrn_ = std::move(hrn); }
+
+  /**
+   * @brief Sets the version.
+   *
+   * @param version.
+   */
+  void SetVersion(int64_t version) { version_ = version; }
+
+  /**
+   * @brief Sets the direct status.
+   *
+   * @param direct status.
+   */
+  void SetDirect(bool direct) { direct_ = direct; }
+
+  /**
+   * @brief Gets the hrn.
+   *
+   * @return The hrn.
+   */
+  const std::string& GetHrn() const { return hrn_; }
+
+  /**
+   * @brief Gets the version.
+   *
+   * @return The version.
+   */
+  int64_t GetVersion() const { return version_; }
+
+  /**
+   * @brief Gets the direct status.
+   *
+   * @return The direct status.
+   */
+  bool IsDirect() const { return direct_; }
+
+ private:
+  std::string hrn_;
+  int64_t version_;
+  bool direct_;
+};
+
+/**
+ * @brief Represents a catalog version
+ */
+class DATASERVICE_READ_API CatalogVersion final {
+ public:
+  /**
+   * @brief Sets the vector of catalog version dependencies.
+   *
+   * @param vector of catalog version dependencies.
+   */
+  void SetCatalogVersionDependencies(
```

Comment:
Please follow naming from spec.
There you have the following fields:
- dependencies
- partitionCounts
- timestamp
- version

Means the getters/setters need to have the same name:
- `GetDependencies` / `SetDependecies`
- `GetPartitionCounts` / `SetPartitionCounts`
and so on.